tests/integration_tests/vectorstores/test_docarray_hnsw_search.py

```diff
@@ -0,0 +1,151 @@
+from pathlib import Path
+
+import numpy as np
+import pytest
+
+from langchain.schema import Document
+from langchain.vectorstores.docarray import DocArrayHnswSearch
+from tests.integration_tests.vectorstores.fake_embeddings import FakeEmbeddings
+
+
+def test_docarray_hnsw_search_vec_store_from_texts(tmp_path: Path) -> None:
+    """Test end to end construction and simple similarity search."""
+    texts = ["foo", "bar", "baz"]
+    docsearch = DocArrayHnswSearch.from_texts(
+        texts,
+        FakeEmbeddings(),
+        work_dir=str(tmp_path),
+        n_dim=10,
+        dist_metric="cosine",
+    )
+    assert isinstance(docsearch, DocArrayHnswSearch)
+    assert docsearch.doc_index.num_docs() == 3
+
+
+def test_docarray_hnsw_search_vec_store_add_texts(tmp_path: Path) -> None:
+    """Test end to end construction and simple similarity search."""
+    docsearch = DocArrayHnswSearch(
+        work_dir=str(tmp_path),
+        n_dim=10,
+        embedding=FakeEmbeddings(),
+        dist_metric="cosine",
+    )
+    assert isinstance(docsearch, DocArrayHnswSearch)
+    assert docsearch.doc_index.num_docs() == 0
+
+    texts = ["foo", "bar", "baz"]
+    docsearch.add_texts(texts=texts)
+    assert docsearch.doc_index.num_docs() == 3
+
+
+@pytest.mark.parametrize("metric", ["cosine", "l2"])
+def test_sim_search(metric: str, tmp_path: Path) -> None:
+    """Test end to end construction and simple similarity search."""
+    texts = ["foo", "bar", "baz"]
+    hnsw_vec_store = DocArrayHnswSearch.from_texts(
+        texts,
+        FakeEmbeddings(),
+        work_dir=str(tmp_path),
+        n_dim=10,
+        dist_metric=metric,
+    )
+    output = hnsw_vec_store.similarity_search("foo", k=1)
+    assert output == [Document(page_content="foo")]
+
+
+@pytest.mark.parametrize("metric", ["cosine", "l2"])
+def test_sim_search_all_configurations(metric: str, tmp_path: Path) -> None:
+    """Test end to end construction and simple similarity search."""
+    texts = ["foo", "bar", "baz"]
+    hnsw_vec_store = DocArrayHnswSearch.from_texts(
+        texts,
+        FakeEmbeddings(),
+        work_dir=str(tmp_path),
+        dist_metric=metric,
+        n_dim=10,
+        max_elements=8,
+        index=False,
```

Comment:
we shouldn't ever be setting this to False right? if im understanding correctly, that prevents the embeddings from being indexed? in which case should we just remove that parameter from constructor altogether and always set it to True?